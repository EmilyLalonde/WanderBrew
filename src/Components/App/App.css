.map-nav {
  margin-bottom: 3%;
}

.map {
  margin-top: 1%;
}

a.Nav {
  color: black;
  font-size: 20px;
  
}

.name-nav {
  text-align: center;
  margin-top: 15px;
}