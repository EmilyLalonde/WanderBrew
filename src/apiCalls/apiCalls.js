export const getBreweriesByState = async (state) => {
  const stateBreweries = `https://api.openbrewerydb.org/breweries?by_state=${state}&per_page=50`
  const response = await fetch(stateBreweries)
  if(!response.ok) {
    throw new Error('There was an error getting your albums')
  }
  const allStateBreweries = await response.json();
  return allStateBreweries
}

// export const getBreweriesByCity = async (city) => {
//   const cityBreweries = `https://api.openbrewerydb.org/breweries?by_name=${city}&per_page=50`
//   const response = await fetch(cityBreweries)
//   if(!response.ok) {
//     throw new Error('There was an error getting your albums')
//   }
//   const allCityBreweries = await response.json();
//   return allCityBreweries
// }


