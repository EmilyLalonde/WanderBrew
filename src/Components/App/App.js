import React, { Component } from "react";
import { Route } from "react-router-dom";
import { connect } from 'react-redux';
import "./App.css";
import MapContainer from "../MapContainer/MapContainer";
import LandingContainer from "../LandingContainer/LandingContainer"
import StateSearchContainer from "../StateSearchContainer/StateSearchContainer"
import NameSearchContainer from "../NameSearchContainer/NameSearchContainer"
import NavContainer from '../NavContainer/NavConatiner'
import StateContainer from '../StateContainer/StateContainer'
import Card from '../Card/Card'
import HomeContainer from '../HomeContainer/HomeContainer'

class App extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    const { stateResults, nameResults} = this.props
    console.log(stateResults)
    return (
      <div>
      <Route
        exact
        path="/landing"
        render={() => (
          <div>
            <LandingContainer />
          </div>
        )}
      />
      <Route
        exact
        path="/"
        render={() => (
          <div>
            <HomeContainer />
          </div>
        )}
      />
      <Route
        exact
        path="/state-search"
        render={() => (
          <div>
            <NavContainer />
            <StateSearchContainer />
          </div>
        )}
      />
      <Route
        exact
        path="/name-search"
        render={() => (
          <div>
            <NavContainer />
            <NameSearchContainer />
          </div>
        )}
      />
      <Route
        exact
        path="/map"
        render={() => (
          <div>
            <NavContainer />
            <MapContainer />
          </div>
        )}
      />
      <Route
          path="/state-search/:id"
          render={({ match }) => {
            console.log('stateResultsFind', stateResults)
            const foundState = stateResults.find(
              state => state.id === parseInt(match.params.id)
            );
              return (
                <div>
                  <Card {...foundState}/>
                </div>
              );
          }} 
        />
        <Route
          path="/name-search/:id"
          render={({ match }) => {
            console.log('stateResultsFind', stateResults)
            const foundName = nameResults.find(
              name => name.id === parseInt(match.params.id)
            );
              return (
                <div>
                  <Card {...foundName}/>
                </div>
              );
          }} 
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  stateResults: state.stateResults,
  nameResults: state.nameResults
})

export default connect(mapStateToProps, null)(App);
