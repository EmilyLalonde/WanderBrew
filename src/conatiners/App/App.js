import React, { Component } from "react";
import { Route } from "react-router-dom";
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { getPopularDenverBreweries, getPopularBreweriesNational } from '../../apiCalls/apiCalls'
import { getDenverBreweries, getNationalBreweries} from '../../actions'
import "./App.css";
import MapContainer from "../../Components/MapContainer/MapContainer";
import StateSearchContainer from "../StateSearchContainer/StateSearchContainer"
import NameSearchContainer from "../NameSearchContainer/NameSearchContainer"
import NavContainer from '../../Components/NavContainer/NavConatiner'
import Card from '../../Components/Card/Card'
import HomeContainer from '../../conatiners/HomeContainer/HomeContainer'
import ImagesContainer from '../../Components/ImagesContainer/ImagesContainer'
import NotesContainer from '../../Components/NotesConatiner/NotesContainer'

export class App extends Component {
  constructor() {
    super();
    this.state = {
      denverBreweries: [],
      nationalBreweries: [],
      notes: []
    }
    this.addNote = this.addNote.bind(this);
  }

  componentDidMount () {
    const { getDenverBreweries, getNationalBreweries} = this.props
    getPopularDenverBreweries()
    .then(data => getDenverBreweries(data))
    getPopularBreweriesNational()
    .then(data => getNationalBreweries(data))
}

  addNote (newNote) {
    this.setState({notes: [...this.state.notes, newNote]})
} 

  render() {
    const { stateResults, nameResults} = this.props
    return (
      <div>
      <Route
        exact
        path="/"
        render={() => (
          <div>
            <NavContainer />
            <HomeContainer/>
          </div>
        )}
      />
      <Route
        exact
        path="/state-search"
        render={() => (
          <div>
          <NavContainer />
            <StateSearchContainer/>
          </div>
        )}
      />
      <Route
        exact
        path="/name-search"
        render={() => (
          <div>
            <NavContainer />
            <NameSearchContainer/>
          </div>
        )}
      />
      <Route
        exact
        path="/map"
        render={() => (
          <div>
            <div className="map-nav">
            <NavContainer />
            </div>
            <div className="map">
            <MapContainer />
            </div>
          </div>
        )}
      />
      <Route
          path="/state-search/:id"
          render={({ match }) => {
            const foundState = stateResults.find(
              state => state.id === parseInt(match.params.id)
            );
              return (
                <div>
                  <div className="name-nav">
                  <NavLink to='/' className='Nav'>-- Home --  </NavLink>
                  </div>
                  <Card {...foundState}/>
                  <ImagesContainer />
                  <NotesContainer addNote={this.addNote} notes={this.state.notes}/>
                </div>
              );
          }} 
        />
        <Route
          path="/name-search/:id"
          render={({ match }) => {
            const foundName = nameResults.find(
              name => name.id === parseInt(match.params.id)
            );
              return (
                <div>
                  <div className="name-nav">
                  <NavLink to='/' className='Nav'>-- Home --</NavLink>
                  </div>
                  <Card {...foundName}/>
                  <ImagesContainer />
                  <NotesContainer addNote={this.addNote} notes={this.state.notes}/>
                </div>
              );
          }} 
        />
        <Route
          path="/home/:id"
          render={({ match }) => {
            const foundDefaultCO = this.props.getDenverBreweries.find(
              brew => brew.id === parseInt(match.params.id)
            );
            const foundDefaultNat = this.props.getNationalBreweries.find(
              brew => brew.id === parseInt(match.params.id)
            );
              return (
                <div>
                  <div className="name-nav">
                  <NavLink to='/' className='Nav'> --Home --</NavLink>
                  </div>
                  <Card {...foundDefaultCO || foundDefaultNat}/>
                  <ImagesContainer />
                  <NotesContainer addNote={this.addNote} notes={this.state.notes}/>
                </div>
              );
          }} 
        />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  stateResults: state.stateResults,
  nameResults: state.nameResults,
  denverResults: state.denverResults,
  nationalResults: state.nationalResults
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({ getDenverBreweries, getNationalBreweries}, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App);
