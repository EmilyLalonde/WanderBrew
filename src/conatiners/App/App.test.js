import React from 'react';
import { shallow } from 'enzyme'
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getBreweriesByState, getBreweriesByName, getPopularDenverBreweries, getPopularBreweriesNational } from "../../apiCalls/apiCalls";

jest.mock('../../apiCalls/apiCalls');

getBreweriesByState.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

getBreweriesByName.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

getPopularDenverBreweries.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

getPopularBreweriesNational.mockImplementation(() => {
  return Promise.resolve({
    id: 1516, 
    name: 'Ratio Beerworks', 
    street: '2920 Larimer St', 
    city: 'Denver', 
    state:'Colorado', 
    phone:'3012669264'
  })
});

describe('App', () => {
  let wrapper;

  let mockStateResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockNameResults = [
    {
      id: 1516, 
      name: 'Ratio Beerworks', 
      street: '2920 Larimer St', 
      city: 'Denver', 
      state:'Colorado', 
      phone:'3012669264'
    }
  ]

  let mockState = {
    stateResults: [mockStateResults],
    nameResults: [mockNameResults]
  }

  let mockAddNote = jest.fn()
  let mockGetPopularDenverBreweries = jest.fn()
  let mockGetPopularBreweriesNational = jest.fn()

  beforeEach(() => {
    wrapper= shallow(<App 
      getPopularDenverBreweries={mockGetPopularDenverBreweries}
      getPopularBreweriesNational={mockGetPopularBreweriesNational}
      addNote={mockAddNote}
    />)
  })

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})

  it('should call mapStateToProps with the props it needs', () => {
    let mockStateResults = [
      {
        id: 1516, 
        name: 'Ratio Beerworks', 
        street: '2920 Larimer St', 
        city: 'Denver', 
        state:'Colorado', 
        phone:'3012669264'
      }
    ]
  
    let mockNameResults = [
      {
        id: 1516, 
        name: 'Ratio Beerworks', 
        street: '2920 Larimer St', 
        city: 'Denver', 
        state:'Colorado', 
        phone:'3012669264'
      }
    ]
  
    let mockState = {
      stateResults: [mockStateResults],
      nameResults: [mockNameResults]
    }

    const expected = {
      stateResults: [mockStateResults],
      nameResults: [mockNameResults],
    }
    const mappedProps = mapStateToProps(mockState)
    
    expect(mappedProps).toEqual(expected);
  })

  it('should call getPopularDenverBreweries and getPopularBreweriesNational when mounting', () => {
    expect(getPopularDenverBreweries).toHaveBeenCalled();
    expect(getPopularBreweriesNational).toHaveBeenCalled();
  });