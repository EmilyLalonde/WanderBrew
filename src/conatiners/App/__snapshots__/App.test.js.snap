// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the data passed through 1`] = `
<div>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/state-search"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/name-search"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/map"
    render={[Function]}
  />
  <Route
    path="/state-search/:id"
    render={[Function]}
  />
  <Route
    path="/name-search/:id"
    render={[Function]}
  />
  <Route
    path="/home/:id"
    render={[Function]}
  />
</div>
`;
